src/proxy/protocol.rs

```diff
@@ -17,6 +18,7 @@ impl Protocol {
         // http2 is easiest to detect
         if bytes.len() >= H2_PREFACE.len() {
             if &bytes[..H2_PREFACE.len()] == H2_PREFACE {
+                println!("^^^^^ is H2");
```

Comment:
I'll remove the `println`s later. This PR is for reviewing purpose only.

Reply:
It's absolutely fine to use `println!` when debugging. If we want to print out diagnostic information that we expect to be useful in the future and don't intend to remove, though, we prefer to use the [`log` crate's macros](https://docs.rs/log/0.4.6/log/#macros), which will log records that include metadata and can be filtered, rather than printing directly to stdout.